search/includes/classes/class-search.php

```diff
@@ -765,6 +770,63 @@ public function filter__ep_do_intercept_request( $request, $query, $args, $failu
 		}
 	}
 
+	/**
+	 * We want to ensure that index exists before doing certain type of operation on it.
+	 *
+	 * The reason is that indexing a document, before mappings were put, would result in index with default, incorrect mapping.
+	 * This method avoids this issue by creating the index with correct mapping if the index doesn't exist yet.
+	 */
+	public function ensure_index_existance( $url, $args ) {
+		$method = strtoupper( $args['method'] ?? '' );
+		$methods_that_need_index = [ 'POST', 'PUT' ];
+		if ( ! in_array( $method, $methods_that_need_index ) ) {
+			// bailing out on methods that would not create index with incorrect mapping
+			return true;
+		}
+
+		$index_name = $this->get_index_name_for_url( $url );
+		if ( ! $index_name ) {
+			return true;
+		}
+
+		$path_parts = explode( '/', trim( $url ) );
+		if ( end( $path_parts ) == $index_name ) {
```

Comment:
This can be swapped with `wp_endswith`

Reply:
This covers both `http://.../index_name/` and `http://.../index_name`, I guess I could do `trim( trim( $url ), '/' )` and then `wp_endswith` on that. Seems kinda similar to me. I don't have strong opinions, if you think I should go with wp_endswith instead, I am fine with that.

Reply:
as long as we don't need other `$path_parts` let's do the `wp_endswith( trim( $url, '/'), $index_name )`

Reply:
done